### ISSUE TITLE
Difference between selectGraphQLResultFromTable and context.pgClient.query

### ISSUE THREAD

**Issue by fouadkada** [graphile/postgraphile#1737]
### Summary

I've read the documentation of `selectGraphQLResultFromTable` that are found here https://www.graphile.org/postgraphile/make-extend-schema-plugin/#the-selectgraphqlresultfromtable-helper and I still cannot fully grasp the difference between `selectGraphQLResultFromTable` and `context.pgClient.query`. It's also not clear to me when one should be used instead of the other.

Can someone please dumb it down more for me?

**Comment by benjie** [comment]
selectGraphQLResultFromTable is used to provide the result of your resolver when your field type is an autogenerated type. This is required to have the lookahead system build the dynamic SQL query based on what the user has requested in their GraphQL request. (This dynamic SQL is then executed via context.pgClient)

context.pgClient is just a database client, you can use it to query the database with arbitrary SQL. This is generally useful for the body of custom mutations, but the end result of your mutation may still need to be a selectGraphQLResultsFromTable if you are using an autogenerated type as the result type of the field.

**Comment by fouadkada** [comment]
in this case, in the following example, what's the benefit of using one approach over the other:
```
module.exports = makeExtendSchemaPlugin(({pgSql: sql}) => ({
    typeDefs: gql`
        type NewPersonAddedPayload {
            person: Person
        }

        extend type Subscription {
            newPersonAdded: NewPersonAddedPayload @pgSubscription(topic: "person_inserted")
        }
    `,

    resolvers: {
        NewPersonAddedPayload: {
            async person(
                event,
                _args,
                context,
                {graphile: {selectGraphQLResultFromTable}}
            ) {
                const { rows } = await context.pgClient.query(
                    "select * from learn.person where id = $1", 
                    [event.record.id]
                );
                // const rows = await selectGraphQLResultFromTable(
                //     sql.fragment`learn.person`,
                //     (tableAlias, sqlBuilder) => {
                //         sqlBuilder.where(
                //             sql.fragment`${tableAlias}.id = ${sql.value(event.record.id)}`
                //         );
                //     }
                // );
                return rows[0];
            },
        },
    },
}));
```

I used the `selectGraphQLResultFromTable` approach (which is commented) and the `context.pgClient.query` and they both yield the same result.

**Comment by benjie** [comment]
Issue a query like:

```graphql
subscription {
  newPersonAdded {
    person {
      id
      organization { id name }
    }
  }
}
```

Replace `organization` and its selection set with _any_ relationship, computed column function, or similar autogenerated functionality other than basic columns, and you'll see that it's just `null` in the pgClient version because the dynamic query is not generated/issued, but in the `selectGraphQLResultFromTable` version it will work because the lookahead system is engaged.
